aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -715,6 +716,79 @@ Tensor index_select_backward(const Tensor& grad, IntArrayRef self_sizes, int64_t
   return at::zeros(self_sizes, grad.options()).index_add_(dim, index, grad);
 }
 
+Tensor & index_fill_(Tensor & self, int64_t dim, const Tensor & index, Scalar source) {
+  at::NoNamesGuard guard;
+
+  TORCH_CHECK_INDEX(
+    index.scalar_type() == ScalarType::Long,
+    "index_fill_(): Expected dtype int64 for index.");
+
+  at::assert_no_overlap(self, index);
+  if (at::has_internal_overlap(self) == at::MemOverlap::YES) {
+    TORCH_WARN(
+      "Use of index_fill_ on expanded tensors is deprecated. "
+      "Please clone() the tensor before performing this operation. "
+      "This also applies to advanced indexing e.g. tensor[mask] = scalar");
+  }
+
+  if (!self.is_complex() && source.isComplex()) {
+    TORCH_CHECK(false, "Converting complex Scalar to non-complex type is forbidden");
```

Comment:
Nit: can you insert `index_fill_` into error message text so that it's clear where it's thrown? The message is generic, and would be hard to debug if someone sees it.